libvast/src/system/index.cpp

```diff
@@ -868,20 +868,25 @@ void index_state::decommission_active_partition(
                            partition_synopsis_pair{id, ps});
               unpersisted.erase(id);
               persisted_partitions.emplace(id);
+              self->send_exit(actor, caf::exit_reason::normal);
               if (completion)
                 completion(caf::none);
             },
             [=, this](const caf::error& err) {
               VAST_DEBUG("{} received error for request to persist partition "
                          "{} {}: {}",
                          *self, schema, id, err);
+              unpersisted.erase(id);
+              self->send_exit(actor, err);
               if (completion)
                 completion(err);
             });
       },
       [=, this](caf::error& err) {
         VAST_ERROR("{} failed to persist partition {} {} with error: {}", *self,
                    schema, id, err);
+        unpersisted.erase(id);
+        self->send_exit(actor, err);
```

Comment:
I don't think this is correct: If we fail to persist a partition it is still unpersisted. It makes sense to do this in the success case, but not so much in the error case.